### PR TITLE
gh-84461: Use HOSTRUNNER to run regression tests (GH-93694)

### PR DESCRIPTION
**Pull request by tiran**

### CODE REVIEW

File: Tools/scripts/run_tests.py [hunk 1]
```diff
@@ -19,15 +21,38 @@ def is_multiprocess_flag(arg):
...
     args = [sys.executable,
             '-u',                 # Unbuffered stdout and stderr
             '-W', 'default',      # Warnings set to 'default'
             '-bb',                # Warnings about bytes/bytearray
-            '-E',                 # Ignore environment variables
             ]
 
+    cross_compile = '_PYTHON_HOST_PLATFORM' in os.environ
+    hostrunner = os.environ.get("_PYTHON_HOSTRUNNER")
+    if hostrunner is None:
```
Comment (brettcannon) [inline Tools/scripts/run_tests.py]:
```suggestion
    if (hostrunner := os.environ.get("_PYTHON_HOSTRUNNER")) is None:
```

File: Tools/scripts/run_tests.py [hunk 2]
```diff
@@ -38,16 +63,30 @@ def main(regrtest_args):
     if sys.platform == 'win32':
         args.append('-n')         # Silence alerts under Windows
     if not any(is_multiprocess_flag(arg) for arg in regrtest_args):
-        args.extend(['-j', '0'])  # Use all CPU cores
+        if cross_compile and hostrunner:
+            # For now use only one core for cross compiled builds.
+            # hostrunner can be expensive.
```
Comment (brettcannon) [inline Tools/scripts/run_tests.py]:
```suggestion
            # For now use only one core for cross-compiled builds;
            # hostrunner can be expensive.
```

**Review by brettcannon — APPROVED** [review]

File: Tools/scripts/run_tests.py (hunk 1 above)
Comment (tiran) [reply Tools/scripts/run_tests.py]: Oh, this newfangled features :p

**Comment by ethanhs** [comment]
Looks good to me! Now I will wrap up my buildbot changes :)

**Pull request merged**